autopilot/autopilot.go

```diff
@@ -191,21 +191,31 @@ func (ap *Autopilot) Run() error {
 			}
 
 			// migration
-			ap.m.TryPerformMigrations(ctx, cfg)
+			ap.m.tryPerformMigrations(ctx, cfg)
 		}()
 	}
 }
 
-func (ap *Autopilot) Stop() error {
-	ap.startStopMu.Lock()
-	defer ap.startStopMu.Unlock()
+func (ap *Autopilot) Stop(ctx context.Context) error {
+	doneChan := make(chan struct{})
+	go func() {
```

Comment:
Is this really necessary? The autopilot seems simple enough to not take a long time shutting down and if it does we should be able to find the reason for it.

Reply:
I don't mind removing it but I think it's a bit weird to pass in contexts and then ignore them. I think this boils down to the question of whether we want to have the shutdown functions take a context yes or no. The reason I added them is mostly because of the tracing provider, which takes a context...

We currently pass the background context to it but I think that's just annoying personally. If something takes a context I think we should pass it a context with a sane timeout. There's nothing that prevents us investigating an unclean shutdown in the autopilot because we return a "timed out" error.

Reply:
I don't think spinning up a goroutine here just to avoid having an unused context is worth it.
We don't gain anything over the user manually killing the process after a timeout or docker doing so. It might actually be worse because we take away the chance to print a stack trace. 
If you really want to use the context then I think a better thing to do would be starting an `autopilot.Shutdown` trace at the beginning of the method to give us a historic datapoint on how long it takes to shut down the autopilot.